python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -36,6 +39,7 @@ def __init__(self, config: Mapping[str, str], log):  # pylint: disable=too-many-
         self.connector = config.get("connector", None)
```

Comment:
We should add a check here that the user has provided one of (password) or (private_key, private_key_path, private_key_password) and throw an exception if neither group is provided. 

You could do this with something like

```
check.invariant(config.get("password", None) is not None or (config.get("private_key", None) is not None and config.get("private_key_path", None)  is not None and config.get("private_key_password", None) is not None), "Missing config: Password or private key authentication required")

```

Reply:
we should probably also check that they dont provide both ways of authenticating